htdocs/accountancy/class/accountancyimport.class.php

```diff
@@ -55,21 +85,26 @@ public function computeAmount(&$arrayrecord, $fieldname, &$listfields, &$listval
 			$fieldname = $fieldArr[1];
 		}
 
-		$debit  = floatval(trim($arrayrecord[11]['val']));
-		$credit = floatval(trim($arrayrecord[12]['val']));
+		$debit_index = 11;
+		$credit_index = 12;
+		$debit_val = trim($arrayrecord[$debit_index]['val']);
+		$credit_val = trim($arrayrecord[$credit_index]['val']);
+		$debit  = floatval($debit_val);
+		$credit = floatval($credit_val);
 		if (!empty($debit)) {
 			$amount = $debit;
 		} else {
 			$amount = $credit;
 		}
+		$listvalues[$debit_index] = "'" . $debit . "'";
```

Comment:
Can you test with this code instead
`$listvalues[$debit_index] = "'" . $this->db->escape($debit) . "'";`